test/utility/parse_opts.c

```diff
@@ -26,19 +26,82 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 #include "test_util.h"
+#define DIR_STORE "dir_store"
 
 extern char *__wt_optarg; /* argument associated with option */
 
+/*
+ * parse_tiered_opts --
+ *     Parse command line options for the tiered storage configurations.
+ */
+static inline void
+parse_tiered_opts(int argc, char *const *argv, TEST_OPTS *opts)
+{
+    int index;
+    int number_of_tiered_options;
+
+    number_of_tiered_options = 0;
+    opts->tiered_storage = false;
+
+    for (int i = 1; i < argc; i++) {
+        /* Tiered storage command line options starts with -P. */
+        if (strstr(argv[i], "-P")) {
+            /* Many more options to come here. */
+            if (argv[i][2] == 'o') {
+                if (argv[i + 1] == NULL)
+                    testutil_die(
+                      EINVAL, "%s option requires an argument %s", opts->progname, argv[i]);
+                number_of_tiered_options += 2;
+            } else if (argv[i][2] == 'T') {
+                /* This parsing is different because -PT does not accept any arguments. */
+                ++number_of_tiered_options;
+                opts->tiered_storage = true;
+            }
+        }
+    }
+
+    /* Return from here if tiered arguments are not passed. */
+    if (!opts->tiered_storage)
+        return;
```

Comment:
What happens if `-Po` is passed and `-PT` is not?  I think this is an error, but probably what will happen is that `tiered_storage` is not set, so `__wt_getopts` is called and `-P` is rejected as illegal.  This may be unexpected to a developer using the API.

So maybe better to either: silently turn on tiered storage (that is assume -PT) if any -Pxxx option is specified.  Or else make a check at this point - if `(!opts->tiered_storage && number_of_tiered_options > 0)` then error with an appropriate message.

Reply:
I would prefer the second option to throw an error if -Pxx is passed without -PT. Fixed.